src/asmcnc/apps/systemTools_app/screens/screen_factory_settings.py

```diff
@@ -685,12 +693,20 @@ def full_console_update(self):
 
         def nested_full_console_update(dt):
 
+            # Ensure git repo is good before anything else happens
+            if not self.set.do_git_fsck():
+                message = "git FSCK errors found! repo corrupt."
+                popup_system.PopupFSCKErrors(self.systemtools_sm.sm, self.l, message, self.set.details_of_fsck)
+                self.console_update_button.text = 'Full Console Update (wifi)'
+                return False
+
             if self.set.get_sw_update_via_wifi():
                 self.set.fetch_platform_tags()
                 self.set.update_platform()
             else: 
                 message = "Could not get software update, please check connection."
                 popup_info.PopupWarning(self.systemtools_sm.sm, self.l, message)
+                self.console_update_button.text = 'Full Console Update (wifi)'
```

Comment:
Does there not need to be a line like this in the if case directly above this else case?

Reply:
no, because it will do a full reboot as part of the update :)